### PR TITLE
Better size estimate for samba...

### PR DESCRIPTION
**Pull request by platbr**
When DLE has a subdir, the estimate was taking too much time, because it was executing in ROOT of the share (even using -D subdir), now it will execute only in the subdir.

On the share's root (only to see the total amount of data):

```
/opt/apps/samba4/bin/smbclient //localhost/test  -d 0 -U guest -E -c "archive 0;recurse;du"
Total number of bytes: 144090056
```

Original code of amanda when estimate size of //localhost/test/subdir:

```
/opt/apps/samba4/bin/smbclient //localhost/test  -d 0 -U guest -E -D subdir -c "archive 0;recurse;du"
Total number of bytes: 144090056

```

With this patch when estimate size of //localhost/test/subdir:

```
/opt/apps/samba4/bin/smbclient //localhost/test  -d 0 -U guest -E -D subdir -c "archive 0;recurse;du /subdir"
Total number of bytes: 72045028
```

```
-bash-4.1$ du -s /tmp/test
159832  /tmp/test
-bash-4.1$ du -s /tmp/test/*
79920   /tmp/test/pasta
79908   /tmp/test/subdir
```

Tested with samba3 and samba4, same behavior.

Wagner Caixeta Rodrigues
wagner {dot} caixeta {at} gmail

### CODE REVIEW

**Comment by chassell** [comment]
This change miraculously seems to be applicable today.  Mostly because it's microscopic.

I'd like this to go in ... but I'd like also to have a double-quote (") to help out du so it never lets spaces intervene in a subdir.

File: application-src/amsamba.pl [hunk 1]
```diff
@@ -510,9 +510,9 @@ sub command_estimate {
 	    push @ARGV, "-D", $self->{subdir},
 	}
 	if ($level == 0) {
-	    push @ARGV, "-c", "archive 0;recurse;du";
+	    push @ARGV, "-c", "archive 1;recurse;du /" . $self->{subdir};
```
Comment (UweMenges) [inline application-src/amsamba.pl]: Why this change of "archive 0" to "archive 1"?

Comment (chassell) [reply application-src/amsamba.pl]: I'm not sure why either...

**Comment by chassell** [comment]
I think this needs to be rejected because *supposedly* the "-D" option set just two lines above is able to perform the same act with "du".   If one uses "du *" then the local directory (which is the CWD) comes back pretty accurately.

File: application-src/amsamba.pl (hunk 1 above)
Comment (UweMenges) [reply application-src/amsamba.pl]: I found reasoning in #76.

Comment (chassell) [reply application-src/amsamba.pl]: It doesn't seem to say why the "archive 0" was changed to archive 1, does it?

I put a new pull-req in to use "du *" because that seems to work as intended, recursively.
